odoo/addons/base/models/ir_ui_view.py

```diff
@@ -1447,3 +1447,109 @@ def _validate_module_views(self, module):
+                    _logger.error("issue occurs")
+
+    def get_faulty_views(self,parent_view,view, disabled_views):
+        """ Returns the faulty views
+            To determine the faulty views, we travel the tree recursively.
+            Disable all children, then enable one by one. If there is an issue.
+            The activated node or at least of its children are faulty.
+            Disable faulty node and continue recursion.
+            This function is never called from odoo code but allow to be called from a server action.
+            Please don't remove
+        """
+        if parent_view == view:
+            self._cr.execute('SAVEPOINT test_fix_views')
+            temp_disabled_views= self.deactivate_lvl1_children(view)
+            if self.view_is_valid(parent_view):
+                if self.has_active_children(view) or len(temp_disabled_views)>1 :
+                    for child_view in (temp_disabled_views):
+                        v =self.get_faulty_views(parent_view,child_view,disabled_views)
+                        for vv in v:
+                            if vv not in disabled_views:
+                                disabled_views+=vv
+                else:
+                    view.active=False
+                    disabled_views+=view #this is the faulty view 
+            else:
+                view.active=False
+                disabled_views+=view
+            self._cr.execute('ROLLBACK TO test_fix_views')
+            return disabled_views
+        else:
+            assert self.view_is_valid(parent_view)
+            view.active=True
+            if self.view_is_valid(parent_view):
+                return disabled_views
+            else:
+                if self.has_active_children(view):
+                    temp_disabled_views= self.deactivate_lvl1_children(view)
+                    if self.view_is_valid(parent_view):
+                        # problem is in one of the children
+                        for child_view in (temp_disabled_views):
+                            v =self.get_faulty_views(parent_view,child_view,disabled_views)
+                            for vv in v:
+                                if vv not in disabled_views:
+                                    disabled_views+=vv
+                        return disabled_views
+                    else:
+                        view.active=False
+                        disabled_views+=view
+                        return disabled_views
+                else:
+                    view.active=False
+                    disabled_views+=view
+                    return disabled_views
+
+    def deactivate_lvl1_children(self, view):
+        """ deactivate and return views that has been deactivated taking care of view priority 
+        NOTE: Do not use self due to recursivity"""
+        temp_disabled_views= self.get_lvl1_children(view)
+        for v in temp_disabled_views:
+            v.active=False
```

Comment:
`temp_disabled_views` should be a recordset, and this should be a single `write` call.